boards/arm/pico_pi_m4/pinmux.c

```diff
@@ -0,0 +1,187 @@
+		IOMUXC_SW_MUX_CTL_PAD_UART1_TX_DATA_MUX_MODE(1) |
+		IOMUXC_SW_MUX_CTL_PAD_UART1_TX_DATA_SION_MASK;
+
+	IOMUXC_I2C1_SCL_SELECT_INPUT = IOMUXC_I2C1_SCL_SELECT_INPUT_DAISY(0);
+	IOMUXC_I2C1_SDA_SELECT_INPUT = IOMUXC_I2C1_SDA_SELECT_INPUT_DAISY(0);
+
+	IOMUXC_SW_PAD_CTL_PAD_UART1_RX_DATA =
+		IOMUXC_SW_PAD_CTL_PAD_UART1_RX_DATA_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_UART1_RX_DATA_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_UART1_RX_DATA_DSE(0)   |
+		IOMUXC_SW_PAD_CTL_PAD_UART1_RX_DATA_HYS_MASK;
+
+	IOMUXC_SW_PAD_CTL_PAD_UART1_TX_DATA =
+		IOMUXC_SW_PAD_CTL_PAD_UART1_TX_DATA_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_UART1_TX_DATA_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_UART1_TX_DATA_DSE(0)   |
+		IOMUXC_SW_PAD_CTL_PAD_UART1_TX_DATA_HYS_MASK;
+ #endif  /* CONFIG_I2C_1 */
+
+
+ #ifdef CONFIG_I2C_2
+	IOMUXC_SW_MUX_CTL_PAD_I2C2_SCL =
+		IOMUXC_SW_MUX_CTL_PAD_I2C2_SCL_MUX_MODE(0) |
+		IOMUXC_SW_MUX_CTL_PAD_I2C2_SCL_SION_MASK;
+	IOMUXC_SW_MUX_CTL_PAD_I2C2_SDA =
+		IOMUXC_SW_MUX_CTL_PAD_I2C2_SDA_MUX_MODE(0) |
+		IOMUXC_SW_MUX_CTL_PAD_I2C2_SDA_SION_MASK;
+
+	IOMUXC_I2C2_SCL_SELECT_INPUT = IOMUXC_I2C2_SCL_SELECT_INPUT_DAISY(1);
+	IOMUXC_I2C2_SDA_SELECT_INPUT = IOMUXC_I2C2_SDA_SELECT_INPUT_DAISY(1);
+
+	IOMUXC_SW_PAD_CTL_PAD_I2C2_SCL =
+		IOMUXC_SW_PAD_CTL_PAD_I2C2_SCL_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_I2C2_SCL_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_I2C2_SCL_DSE(0)   |
+		IOMUXC_SW_PAD_CTL_PAD_I2C2_SCL_HYS_MASK;
+
+	IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA =
+		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_DSE(0)   |
+		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_HYS_
+ #endif  /* CONFIG_I2C_2 */
+
+
+ #ifdef CONFIG_I2C_3
+	IOMUXC_SW_MUX_CTL_PAD_I2C3_SCL =
+		IOMUXC_SW_MUX_CTL_PAD_I2C3_SCL_MUX_MODE(0) |
+		IOMUXC_SW_MUX_CTL_PAD_I2C3_SCL_SION_MASK;
+	IOMUXC_SW_MUX_CTL_PAD_I2C3_SDA =
+		IOMUXC_SW_MUX_CTL_PAD_I2C3_SDA_MUX_MODE(0) |
+		IOMUXC_SW_MUX_CTL_PAD_I2C3_SDA_SION_MASK;
+
+	IOMUXC_I2C3_SCL_SELECT_INPUT = IOMUXC_I2C3_SCL_SELECT_INPUT_DAISY(2);
+	IOMUXC_I2C3_SDA_SELECT_INPUT = IOMUXC_I2C3_SDA_SELECT_INPUT_DAISY(2);
+
+	IOMUXC_SW_PAD_CTL_PAD_I2C3_SCL =
+		IOMUXC_SW_PAD_CTL_PAD_I2C3_SCL_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_I2C3_SCL_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_I2C3_SCL_DSE(0)   |
+		IOMUXC_SW_PAD_CTL_PAD_I2C3_SCL_HYS_MASK;
+
+	IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA =
+		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_DSE(0)   |
+		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_HYS_MASK;
+ #endif  /* CONFIG_I2C_3 */
+
+
+ #ifdef CONFIG_I2C_4
+	IOMUXC_SW_MUX_CTL_PAD_SAI1_RX_SYNC =
+		IOMUXC_SW_MUX_CTL_PAD_SAI1_RX_SYNC_MUX_MODE(3) |
+		IOMUXC_SW_MUX_CTL_PAD_SAI1_RX_SYNC_SION_MASK;
+	IOMUXC_SW_MUX_CTL_PAD_SAI1_RX_BCLK =
+		IOMUXC_SW_MUX_CTL_PAD_SAI1_RX_BCLK_MUX_MODE(3) |
+		IOMUXC_SW_MUX_CTL_PAD_SAI1_RX_BCLK_SION_MASK;
+
+	IOMUXC_I2C4_SCL_SELECT_INPUT = IOMUXC_I2C4_SCL_SELECT_INPUT_DAISY(3);
+	IOMUXC_I2C4_SDA_SELECT_INPUT = IOMUXC_I2C4_SDA_SELECT_INPUT_DAISY(3);
```

Comment:
You seem to be mixing I2C and SAI within this `#ifdef CONFIG_I2C_4` block. Is that intended?

Reply:
I looked carefully at the pinmux, I took it from pinmux made for FreeRTOS: https://github.com/TechNexion/freertos-tn/blob/freertos_1.0.1_imx7d/examples/imx7d_pico_m4/pin_mux.c

It does not seem to me that there is an error, I tried to assign the pins like the one on the WaRP7 and the serial debug did not work

Reply:
You are right, my mistake

Reply:
No problem, Thanks